sdk/tables/azure-data-tables/azure/data/tables/aio/_base_client_async.py

```diff
@@ -35,17 +35,20 @@
 from .._sdk_moniker import SDK_MONIKER
 from ._policies_async import AsyncTablesRetryPolicy
 
+if TYPE_CHECKING:
+    from azure.core.credentials_async import AsyncTokenCredential
+
 
 class AsyncTablesBaseClient(AccountHostsMixin):
 
     def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
         self,
         endpoint: str,
         *,
-        credential: Optional[Union[AzureSasCredential, AzureNamedKeyCredential]] = None,
+        credential: Optional[Union[AzureSasCredential, AzureNamedKeyCredential, "AsyncTokenCredential"]] = None,
         **kwargs: Any
     ) -> None:
-        super(AsyncTablesBaseClient, self).__init__(endpoint, credential=credential, **kwargs)
+        super(AsyncTablesBaseClient, self).__init__(endpoint, credential=credential, **kwargs)  # type: ignore
```

Comment:
Is this a problem that we are passing an async credential policy into the parent base_client constructor?

Reply:
Yes, this gets passed all the way down to `AccountHostsMixin` which expects a sync token credential.